src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -98,7 +98,8 @@ export interface InitialState {
   keyring: Keyring
   adapters: Adapters | null
   wallet: HDWallet | null
-  type: KeyManager | null
+  modalType: KeyManager | null
+  connectedType: KeyManager | null
```

Comment:
The main change in this PR: separating the "what's connected" from "what set of modals are we showing".